gpflow/optimizers/natgrad.py

```diff
@@ -70,18 +69,22 @@ def naturals_to_xi(self, nat1, nat2):
         :param nat2: the θ₂ parameter
         :return: tuple `xi1`, `xi2`
         """
-        pass  # pragma: no cover
 
 
 class NaturalGradient(tf.optimizers.Optimizer):
-    def __init__(self, gamma: Scalar, name=None):
+    def __init__(self, gamma: Scalar, xi_transform: XiTransform = XiNat(), name=None):
```

Comment:
Custom transformations go next to parameters and the can be different for a set of parameters. You cannot pass it in `init` method.

Reply:
this is a default, it seems

Reply:
see line 101

Reply:
This is simply changing the default when for a parameter you haven't passed in a transformation.

Reply:
Warning, this line will instantiate a static variable `XiNat` which will be shared by *all* instances of `NaturalGradient`.

I suggest changing to a class and instantiating.

```suggestion
    def __init__(self, gamma: Scalar, default_xi_transform: XiTransform = XiNat, name=None):
```
I know that currently the XiTransforms are stateless, but the abstractmethod APIs are not (they take self/arent classmethods)! So either we should say: you can define your own XiTransform which can take state (in which case, we should change this API), or you can not define your own stateful XiTransform  (and that should be reflected in API).

I suggest former.

Reply:
the XiNat transform does not have a state so it makes no difference if its instance is shared.

Reply:
We can change the XiTransforms to be classmethods

Reply:
```suggestion
    def __init__(self, gamma: Scalar, xi_transform: XiTransform = XiNat, name=None):
```

Reply:
I'm sorry, but I still don't understand what transform does in `init`. It is "per parameter set" attribute. The default is `XiNat`, and if somebody wants to change it, he/she would need to specify it explicitly via parameters settings.

Reply:
What Artem want's to point out is that you may pass in let's say variational distributions of 3 layers (at least you could do that in the old code I think), and for some reason you want different transformation to be used for some of the parameters. You don't seem to be able to do that now. In general, the transformation should be tied to the parameter and not the optimizer

Reply:
You can still do that, just as before. I've not changed the interface (just minorly added to it, as I needed that for downstream code).

Reply:
@st--, I think what I wanted to know is "why are you changing it"?